src/github.com/getlantern/peerscanner/web.go

```diff
@@ -149,12 +149,29 @@ func clientIpFor(req *http.Request, name string) string {
 	ip := strings.TrimSpace(ips[0])
 	// TODO: need a more robust way to determine when a non-fallback host looks
 	// like a fallback.
-	if !isFallback(name) && strings.HasPrefix(ip, "128.199") {
+	hasFallbackIp := isFallbackIp(ip)
+	if !isFallback(name) && hasFallbackIp {
 		log.Errorf("Found fallback ip %v for non-fallback host %v", ip, name)
 		return ""
-	} else if isFallback(name) && !strings.HasPrefix(ip, "128.199") {
+	} else if isFallback(name) && !hasFallbackIp {
 		log.Errorf("Found non-fallback ip %v for fallback host %v", ip, name)
 		return ""
 	}
 	return ip
 }
+
+func isFallbackIp(ip string) bool {
+	for _, prefix := range fallbackIPPrefixes {
+		if strings.HasPrefix(ip, prefix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+var fallbackIPPrefixes = []string{
+	"128.199",
+	"178.62",
+	"188.166",
+}
```

Comment:
>>  Something would need to be done for peerscanner startup to decide what to do in the gap until servers first register. One option is that on startup peerscanner just asks all servers with fallback-like names to register, and it  doesn't go into the main loop until all have done so or there's some timeout.
>
>We're already in a state where servers are registered. On startup, peerscanner pulls all that information in from CloudFlare and as long as the already registered servers pass the test, they remain in rotation.

I'm not sure I said it right..  I was explaining a problem with my proposal that servers provide this token upon registering: that on peerscanner startup, there is a gap of time when peerscanner can learn about server names and IPs, but not about whether these are real fallbacks (unless we find some way to store this info somewhere convenient enough).

> Mainly this is to prevent a situation that we saw occasionally in which a peer would register via using a call proxied via a fallback.

I don't understand.  In this scenario I'd imagine the server name is more reliable than the IP? How would the fallback in the middle get to mess with the name?